x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/utils.test.ts

```diff
@@ -21,3 +23,66 @@ describe('extractDomainAndEntryPointFromUrl', () => {
     });
   });
 });
+
+describe('getDomainWithProtocol', () => {
+  const { http } = mockHttpValues;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('passes through domain if it starts with https', async () => {
+    const result = await getDomainWithProtocol('https://elastic.co');
+
+    expect(result).toEqual('https://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(0);
+  });
+
+  it('passes through domain if it starts with http', async () => {
+    const result = await getDomainWithProtocol('http://elastic.co');
+
+    expect(result).toEqual('http://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(0);
+  });
+
+  it('returns domain with https protocol if back-end valdates https', async () => {
```

Comment:
test name suggestion
```suggestion
  it('returns domain with https protocol if the back-end validates https', async () => {
```